packages/relay/src/lib/types/events.ts

```diff
@@ -37,4 +38,5 @@ export interface IExecuteQueryEventPayload {
   interactingEntity: string;
   status: string;
   requestId: string;
+  originalCallerAddress: string | undefined;
```

Comment:
When requests are made through queries that do not always involve `originalCaller`, as explained in the PR's description.

Reply:
Is this used only for logs?

Reply:
No it's the actual ethAddress we use for the spanding plans

Reply:
I was thinking, instead of tracking the remaining total budget in-memory inside `HbarLimitService` (we would have to divide it by the number of Kubernetes instances, etc - it is creating unnecessary complexity in the configurations), we could create a new subscription tier (`OPERATOR`) and add a pre-configured spending plan which links the ETH address of the relay operator account to it.

In this way we can only specify the total limit for the `OPERATOR` tier and it will be shared by all instances of the relay. 

What do you think?

Reply:
Ofc, this could be handled in a separate PR as an improvement. We can discuss it today in the parking lot of the daily meeting.

Reply:
Also, here we would be passing the ETH address of the operator account instead of `undefined` for these queries executed by the operator.

Reply:
> Also, here we would be passing the ETH address of the operator account instead of undefined for these queries executed by the operator.

@victor-yanev So you're saying that we would limit Hbar spending on the operator as well?

Reply:
@victor-yanev Good suggestion! Please open a ticket and move the suggestion there so we can come back to improve the hbar limit service.

Reply:
https://github.com/hashgraph/hedera-json-rpc-relay/issues/3099